### PR TITLE
Add static analysis workflow to the ci

### PR DESCRIPTION
**Pull request by lioramilbaum**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,13 @@ logs: up
...
 	pip install -r tests/requirements.txt
 
-test: dependencies up
+test: functional static_analysis
+
+functional: dependencies up
 	pytest --noconftest tests/functional
+
+static_analysis: dependencies
+	flake8
```
Comment (mprahl) [inline Makefile:26]: Could we wait for #290 to get merged so we can use tox to run `flake8`?

Comment (lioramilbaum) [reply Makefile:26]: We can or we can merge this one and do the requested change after.

File: .github/workflows/static_analysis.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.9
+      - name: systic_analysis
```
Comment (mprahl) [inline .github/workflows/static_analysis.yml:17]:
```suggestion
      - name: static_analysis
```

**Comment by mprahl** [comment]
Thanks @lioramilbaum. After the typo, we can go ahead and merge this.

File: Makefile [hunk 3]
```diff
@@ -14,7 +14,13 @@ logs: up
...
+	pip install -r requirements.txt
 	pip install -r tests/requirements.txt
 
-test: dependencies up
+test: functional static_analysis
+
+functional: dependencies up
 	pytest --noconftest tests/functional
+
+static_analysis: dependencies
```
Comment (mprahl) [inline Makefile:25]: Could you add how to run this in the read me please?

Comment (lioramilbaum) [reply Makefile:25]: Done

**Pull request merged**